server/router/api/v1/user_service.go

```diff
@@ -196,11 +201,17 @@ func (s *APIV1Service) UpdateUser(ctx context.Context, request *v1pb.UpdateUserR
 	}
 	for _, field := range request.UpdateMask.Paths {
 		if field == "username" {
+			if workspaceGeneralSetting.DisallowChangeUsername {
+				continue
```

Comment:
Can these checks return an error. if no response is returned the use will never know why their value doesnt get saved

Reply:
Thank you! The latest commit will gray out corresponding field, and returns an error in api instead